test/Java9andUp/src/org/openj9/test/java9AttachAPI/TargetManager.java

```diff
@@ -0,0 +1,412 @@
+	}
+
+	public enum TargetStatus {
+		INIT_SUCCESS, INIT_FAILURE
+	}
+
+	public static void setVerbose(boolean v) {
+		verbose = v;
+	}
+
+	/**
+	 * @param cmdName Classname of the Java program to launch
+	 * @param targetId optional attach API target ID
+	 */
+	public TargetManager(String cmdName, String targetId) {
+		this.targetId = targetId;
+		this.proc = launchTarget(cmdName, targetId, null, null, null);
+	}
+
+	/**
+	 * @param cmdName Classname of the Java program to launch
+	 */
+	public TargetManager(String cmdName) {
+		this.proc = launchTarget(cmdName, null, null, null, null);
+	}
+
+	/**
+	 * @param cmdName Classname of the Java program to launch
+	 * @param targetId optional attach API target ID
+	 * @param appArgs Application arguments
+	 */
+	public TargetManager(String cmdName, String targetId, List<String> appArgs) {
+		this.targetId = targetId;
+		this.proc = launchTarget(cmdName, targetId, null, null, appArgs);
+	}
+
+	/**
+	 * @param cmdName Classname of the Java program to launch
+	 * @param targetId optional attach API target ID
+	 * @param vmArgs VM arguments
+	 * @param appArgs Application arguments
+	 */
+	public TargetManager(String cmdName, String targetId, List<String> vmArgs, List<String> appArgs) {
+		this.targetId = targetId;
+		this.proc = launchTarget(cmdName, targetId, null, vmArgs, appArgs);
+	}
+
+	/**
+	 * @param cmdName Classname of the Java program to launch
+	 * @param targetId optional attach API target ID
+	 * @param displayName
+	 * @param vmArgs VM arguments
+	 * @param appArgs Application arguments
+	 */
+	public TargetManager(String cmdName, String targetId, String displayName, List<String> vmArgs, List<String> appArgs) {
+		this.targetId = targetId;
+		this.proc = launchTarget(cmdName, targetId, displayName, vmArgs,
+				appArgs);
+	}
+
+	/**
+	 * launch a TargetVM process.  Copy the relevant environment such as VM options and 
+	 * classpath to the target's command line.
+	 * @param cmdName Classname of the Java program to launch
+	 * @param targetId optional attach API target ID
+	 * @param myDisplayName  optional attach API target display name
+	 * @param vmArgs VM arguments, such as system property settings
+	 * @param appArgs Application arguments
+	 * @return launched process
+	 */
+	private Process launchTarget(String cmdName, String targetId, String myDisplayName, List<String> vmArgs, List<String> appArgs) {
+		
+		/* Used to build up a list of command line arguments */
+		List<String> argBuffer = new ArrayList<String>();
+		String[] args = {};
+		this.displayName = myDisplayName;
+		Runtime me = Runtime.getRuntime();
+		char fs = File.separatorChar;
+		
+		/* grab the location of the java launcher. */
+		String javaExec = System.getProperty("java.home") + fs + "bin" + fs
+				+ "java";
+		logger.debug("javaExec="+javaExec);
+		
+		/* These are the VM options to the parent process, which we wish to pass on to the child. */
+		String sideCar = System.getProperty("java.sidecar"); 
```

Comment:
This ("java.sidecar") is obsolete, is this entire file an old copy of this helper?

Reply:
Note that this is set by the playlist to indicate VM settings required by child processes (see test/Java8andUp/playlist.xml). Should I use a different system property?

Reply:
Ok, I though it was a reference to something old and obsolete.